### PR TITLE
Jetpack Manage: Creating a child component of SingleSellUpsellLightbox

### PR DESCRIPTION
**Pull request by coder-karen**
Follow-on from https://github.com/Automattic/wp-calypso/pull/86178

## Proposed Changes

* This PR extends the above PR by converting some of the content within the SingleSiteUpsellComponent into a child component.
* This PR also makes sure the 'issue license' button redirects you straight to the add card page, instead of the issue licenses page.
* This PR also fixes a typo in the previous components name.

## Testing Instructions

* Testing instructions should be exactly the same as those listed here: https://github.com/Automattic/wp-calypso/pull/86178
* When testing the issue licenses button, you should now be directed straight to the 'add card' page, not the issue licenses page.

Screenshot 2024-01-12 at 19 40 40

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/jetpack-cloud/sections/partner-portal/license-lightbox/license-lightbox-purchase-via-jetpackcom.tsx [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+	const monthlyPrice = parseFloat( currentProduct.amount );
+
+	let yearlyPrice = 0;
+
+	if ( currentProduct.currency === 'USD' ) {
+		yearlyPrice = monthlyPrice * 12;
+	}
+
+	const learnMoreLink =
+		'https://jetpack.com/support/jetpack-manage-instructions/jetpack-manage-billing-payment-faqs';
```
Comment (tbradsha) [inline client/jetpack-cloud/sections/partner-portal/license-lightbox/license-lightbox-purchase-via-jetpackcom.tsx]: Probably should use `localizeURL()` on this.

Comment (coder-karen) [reply client/jetpack-cloud/sections/partner-portal/license-lightbox/license-lightbox-purchase-via-jetpackcom.tsx]: 👍 Thanks for mentioning that, added now.

File: client/jetpack-cloud/sections/partner-portal/license-lightbox/single-site-upsell-lightbox.tsx [hunk 2]
```diff
@@ -33,21 +30,19 @@ export default function SingleSellUpsellLightbox( {
...
 		onClose?.();
 	}, [ hideLicenseInfo, onClose ] );
 
+	const sites = useSelector( getSites );
+
+	const selectedSite = siteId
+		? sites.find( ( site ) => site?.ID === parseInt( siteId as unknown as string ) )
+		: null;
+
+	const { submitForm } = useSubmitForm( selectedSite );
```
Comment (tbradsha) [inline client/jetpack-cloud/sections/partner-portal/license-lightbox/single-site-upsell-lightbox.tsx:44]: What is this used for?

Comment (coder-karen) [reply client/jetpack-cloud/sections/partner-portal/license-lightbox/single-site-upsell-lightbox.tsx:44]: This is the logic I've used to make sure the Issue License button redirects to the payment card page, following the same logic used on the `partner-portal/licenses` page itself (at `client/jetpack-cloud/sections/partner-portal/issue-license-v2/review-licenses/pricing-summary.tsx`)

File: client/jetpack-cloud/sections/partner-portal/license-lightbox/single-site-upsell-lightbox.tsx [hunk 3]
```diff
@@ -59,29 +54,17 @@ export default function SingleSellUpsellLightbox( {
...
-		dispatch(
-			recordTracksEvent( 'calypso_jetpack_single_site_upsell_proceed_to_checkout_click', {
-				product: currentProduct.slug,
-			} )
-		);
-		onHideLicenseInfo();
-	}, [ currentProduct, dispatch, onHideLicenseInfo ] );
+		submitForm( [
+			{
+				...currentProduct,
```
Comment (tbradsha) [inline client/jetpack-cloud/sections/partner-portal/license-lightbox/single-site-upsell-lightbox.tsx:59]: This didn't add `siteId` to the URL until I added this:

```suggestion
				...currentProduct,
				siteId,
```

Comment (coder-karen) [reply client/jetpack-cloud/sections/partner-portal/license-lightbox/single-site-upsell-lightbox.tsx:59]: Ah, thank you!

File: client/jetpack-cloud/sections/partner-portal/license-lightbox/license-lightbox-purchase-via-jetpackcom.tsx [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+	onClose: () => void;
+	siteId?: number;
+	currentProduct: APIProductFamilyProduct;
+};
+
+const LicenseLightboxPurchaseViaJetpackcom: FunctionComponent< Props > = ( {
+	productSlug,
+	onClose,
+	siteId,
+	currentProduct,
```
Comment (tbradsha) [inline client/jetpack-cloud/sections/partner-portal/license-lightbox/license-lightbox-purchase-via-jetpackcom.tsx]: Technically we don't need to use (and perhaps shouldn't use) `currentProduct`, as there are two different products here, and we're buying the WP.com product (`productSlug`), not the JP Manage product (`currentProduct.slug`).

File: client/jetpack-cloud/sections/partner-portal/license-lightbox/single-site-upsell-lightbox.tsx (hunk 3 above)
Comment (tbradsha) [reply client/jetpack-cloud/sections/partner-portal/license-lightbox/single-site-upsell-lightbox.tsx:59]: Ignore this; probably a Calypso artifact.

File: client/jetpack-cloud/sections/partner-portal/license-lightbox/license-lightbox-purchase-via-jetpackcom.tsx (hunk 4 above)
Comment (coder-karen) [reply client/jetpack-cloud/sections/partner-portal/license-lightbox/license-lightbox-purchase-via-jetpackcom.tsx]: 👍 Thank you

**Review by tbradsha — APPROVED** [review]
Works well!

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/10808243

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @coder-karen for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/10808243

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @coder-karen for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/10808243).

**Pull request merged**